sdk/export/metric/metric.go

```diff
@@ -171,6 +171,82 @@ type Exporter interface {
 	Export(context.Context, CheckpointSet) error
 }
 
+// Convenience function that creates a slice of labels from the passed
+// iterator. The iterator is set up to start from the beginning before
+// creating the slice.
+func IteratorToSlice(iter LabelIterator) []core.KeyValue {
+	l := iter.Len()
+	if l == 0 {
+		return nil
+	}
+	iter.idx = -1
+	slice := make([]core.KeyValue, 0, l)
+	for iter.Next() {
+		slice = append(slice, iter.Label())
+	}
+	return slice
+}
+
+type LabelStorage interface {
+	NumLabels() int
+	GetLabel(int) core.KeyValue
+}
+
+type LabelSlice []core.KeyValue
+
+var _ LabelStorage = LabelSlice{}
+
+func (s LabelSlice) NumLabels() int {
+	return len(s)
+}
+
+func (s LabelSlice) GetLabel(idx int) core.KeyValue {
+	return s[idx]
+}
+
+// LabelIterator allows iterating over ordered set of labels. The
+// typical use of the iterator is as follows:
+//
+//     iter := export.NewLabelIterator(getStorage())
+//     for iter.Next() {
+//       label := iter.Label()
+//       // or, if we need an index:
+//       // idx, label := iter.IndexedLabel()
+//       // do something with label
+//     }
+type LabelIterator struct {
+	storage LabelStorage
+	idx     int
+}
+
+func NewLabelIterator(storage LabelStorage) LabelIterator {
```

Comment:
I like this use of a struct value as the iterator. :+1:

Reply:
Yeah, thanks. :) I like it more too.